bambi/interpret/helpers.py

```diff
@@ -0,0 +1,215 @@
+    TypeError
+        If 'conditional' is a dict and the values are not of type int, float, list, or np.ndarray.
+        If 'conditional' is a list and the elements are not of type str.
+        If 'variable' is a dict and there is more than one key.
+
+    """
+    if variable and effect_type not in ["comparisons", "slopes", "predictions"]:
+        raise ValueError(
+            "'If passing an argument to 'variable', the parameter 'effect_type' must be either "
+            f"'comparisons' or 'slopes'. Received: {effect_type}"
+        )
+
+    if isinstance(conditional, dict):
+        for value in conditional.values():
+            if not isinstance(value, (int, float, list, np.ndarray)):
+                raise TypeError(
+                    "Dictionary values must be of type int, float, list, or np.ndarray. "
+                    f"Received: {type(value)}"
+                )
+
+    if isinstance(conditional, list):
+        for value in conditional:
+            if not isinstance(value, str):
+                raise TypeError(f"Elements of list must be of type str. Received: {type(value)}")
+
+    conditional = ConditionalInfo(model, conditional)
+    kwargs["effect_type"] = effect_type
+
+    if variable:
+        if isinstance(variable, dict):
+            if len(variable) > 1:
+                raise ValueError("Variable dictionary must have only one key.")
+
+        if not eps and effect_type == "comparisons":
+            eps = 0.5
+        elif not eps and effect_type == "slopes":
+            eps = 1e-4
+
+        grid = bool(conditional.covariates)
+        variable = VariableInfo(model, variable, kind=effect_type, eps=eps, grid=grid)
+
+    return create_grid(conditional, variable, **kwargs)
+
+
+def _prepare_idata(idata: InferenceData, data: xr.Dataset) -> InferenceData:
+    """Prepare InferenceData object for use in `select_draws` by removing the
+    'observed_data' group and replacing it with another 'data' group that contains
+    the data used to generate predictions.
+
+    Parameters
+    ----------
+    idata : InferenceData
+        InferenceData object containing the inference data after performing `model.predict`.
+    data : xr.Dataset
+        The Dataset passed as 'data' to `model.predict` to generate predictions.
+
+    Returns
+    -------
+    InferenceData
+        A new InferenceData object with the 'observed_data' group removed and
+        replaced with a 'data' group that contains the 'data' used to generate
+        predictions.
+
+    Raises
+    ------
+    ValueError
+        If the InferenceData object does not contain a 'data' or 'observed_data' group.
+    """
+
+    if "observed_data" in idata.groups():
+        coordinate_name = list(idata["observed_data"].coords)
+        del idata.observed_data
+        idata.add_groups(data=data)
+    elif "data" in idata.groups():
+        coordinate_name = list(idata["observed_data"].coords)
```

Comment:
Is this correct? Does it need to look for the `.coords` in the `observed_data` group or the `data` group?

Reply:
I think we can remove the `elif "data" in idata.groups():` part. I originally had this I believe due to some caching reason. But since I now have 

```python3
  idata = idata.copy()
  ...
  idata_new = _prepare_idata(idata, xr_df)
``
everything is fine (the tests and docs notebook run).